controllers/managedclusterview_controller.go

```diff
@@ -0,0 +1,195 @@
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Successfully reconciled ManagedClusterView", "name", managedClusterView.Name)
+
+	return ctrl.Result{}, nil
+}
+
+func createOrUpdateConfigMap(ctx context.Context, c client.Client, managedClusterView viewv1beta1.ManagedClusterView, logger *slog.Logger) error {
+	logger = logger.With("ManagedClusterView", managedClusterView.Name, "Namespace", managedClusterView.Namespace)
+
+	var resultData map[string]string
+	err := json.Unmarshal(managedClusterView.Status.Result.Raw, &resultData)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal result data: %v", err)
+	}
+
+	clientInfoMap := make(map[string]ClientInfo)
+
+	for _, value := range resultData {
+		var odfInfo ocsv1alpha1.OdfInfoData
+		err := yaml.Unmarshal([]byte(value), &odfInfo)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal ODF info data: %v", err)
+		}
+
+		providerInfo := ProviderInfo{
+			Version:                    odfInfo.Version,
+			DeploymentType:             odfInfo.DeploymentType,
+			CephClusterFSID:            odfInfo.StorageCluster.CephClusterFSID,
+			StorageProviderEndpoint:    odfInfo.StorageCluster.StorageProviderEndpoint,
+			NamespacedName:             odfInfo.StorageCluster.NamespacedName,
+			StorageSystemName:          odfInfo.StorageSystemName,
+			ProviderManagedClusterName: managedClusterView.Namespace,
+		}
+
+		for _, client := range odfInfo.Clients {
+			managedCluster, err := utils.GetManagedClusterById(ctx, c, client.ClusterID)
+			if err != nil {
+				return err
+			}
+			clientInfo := ClientInfo{
+				ClusterID:                client.ClusterID,
+				Name:                     client.Name,
+				ProviderInfo:             providerInfo,
+				ClientManagedClusterName: managedCluster.Name,
+			}
+			clientInfoMap[fmt.Sprintf("%s/%s", managedCluster.Name, client.Name)] = clientInfo
+		}
+	}
+
+	configMapData := make(map[string]string)
+	for clientName, clientInfo := range clientInfoMap {
+		clientInfoJSON, err := json.Marshal(clientInfo)
+		if err != nil {
+			return fmt.Errorf("failed to marshal client info: %v", err)
+		}
+		configMapData[clientName] = string(clientInfoJSON)
+	}
+
+	operatorNamespace := os.Getenv("POD_NAMESPACE")
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ClientInfoConfigMapName,
+			Namespace: operatorNamespace,
+		},
+	}
+
+	op, err := controllerutil.CreateOrUpdate(ctx, c, configMap, func() error {
+		configMap.Data = configMapData
+
+		ownerExists := false
+		for _, ownerRef := range configMap.OwnerReferences {
+			if ownerRef.UID == managedClusterView.UID {
+				ownerExists = true
+				break
+			}
+		}
+
+		if !ownerExists {
+			ownerRef := *metav1.NewControllerRef(&managedClusterView, viewv1beta1.GroupVersion.WithKind("ManagedClusterView"))
+			logger.Info("OwnerRef added", "UID", string(ownerRef.UID))
+			configMap.OwnerReferences = append(configMap.OwnerReferences, ownerRef)
+		}
```

Comment:
This ConfigMap is created in operator namespace, which means the owner also needs to be in the same namespace or be cluster-scoped. MCVs can not be the owner here. ManagedClusters that own the MCVs can be the owner.

Reply:
Done